gprofiler/profilers/python.py

```diff
@@ -119,13 +119,14 @@ def _make_command(self, pid: int, output_path: str) -> List[str]:
             "--full-filenames",
         ]
 
-    def _profile_process(self, process: Process) -> Optional[StackToSampleCount]:
-        try:
-            logger.info(
-                f"Profiling process {process.pid} with py-spy", cmdline=process.cmdline(), no_extra_to_server=True
-            )
-        except NoSuchProcess:
-            return None
+    @staticmethod
+    def _profiling_error_stack(reason: str, comm: str) -> StackToSampleCount:
+        # like _profiling_skipped_stack of java.py
+        return Counter({f"{comm};[Profiling error: {reason}]": 1})
+
+    def _profile_process(self, process: Process) -> StackToSampleCount:
+        logger.info(f"Profiling process {process.pid} with py-spy", cmdline=process.cmdline(), no_extra_to_server=True)
```

Comment:
You removed the `try:` around this, so if `process.cmdline` raises "NoSuchProcess", it will be caught in `profiler_base.py` and just written as log but no indicating stack.

If you move `_profiling_error_stack` to `profiiler_base` than you can just catch all those errors there and write the appropriate stack. It will also catch the other exceptions thrown in this function that didn't return `None` but just raised an exception